# 問4-8 実際のゼロ・クーポン債

問題文にあるような実際の税金の支払い方を行う場合の債券(額面価格$100$)を投資家が購入する時のキャッシュフローは、
\begin{align}
\left(\ -P_0,\ -\frac{100-P_0}{n}t,\ -\frac{100-P_0}{n}t,\ \cdots,\ 100-\frac{100-P_0}{n}t\right)
\end{align}

一方、問題4-7にある債券1と債券2をそれぞれ$\alpha$単位と$\beta$単位づつ購入する場合の投資家のキャッシュフローは、
\begin{align}
\left(\ -(\alpha P_1+\beta P_2),\ (10\alpha+7\beta)(1-t),\ (10\alpha+7\beta)(1-t),\ \cdots,\ 100(\alpha+\beta)-\{100(\alpha+\beta)-(\alpha P_1+\beta P_2)\}t\ \right)
\end{align}

この二つのキャッシュフローが一致するように$\alpha$と$\beta$を選ぶ。初年度初め、$1$〜$9$年後時点、$10$年後でそれぞれ
\begin{eqnarray}
  \begin{cases}
    -P_0 = -(\alpha P_1+\beta P_2) & \\
    -\frac{100-P_0}{n}t = (10\alpha+7\beta)(1-t) & \\
    100-\frac{100-P_0}{n}t = 100(\alpha+\beta)-\{100(\alpha+\beta)-(\alpha P_1+\beta P_2)\}t
  \end{cases}
\end{eqnarray}
を満たせば良い。

$n=10$、$t=0.3$、$P_1=92.21$、$P_2=75.84$を代入する
\begin{eqnarray}
  \begin{cases}
    -P_0 = -(92.21\alpha+75.84\beta) & \\
    -\frac{100-P_0}{10}\times0.3 = (10\alpha+7\beta)(1-0.3) & \\
    100-\frac{100-P_0}{10}\times0.3 = 100(\alpha+\beta)-\{100(\alpha+\beta)-(92.21\alpha+75.84\beta)\}\times0.3
  \end{cases}
\end{eqnarray}

整理する
\begin{eqnarray}
  \begin{cases}
    -P_0+92.21\alpha+75.84\beta = 0 & \\
    0.03P_0 -7\alpha -4.9\beta = 3 & \\
    0.03P_0 -97.663\alpha -92.752\beta = -97
  \end{cases}
\end{eqnarray}

計算する

In [3]:
import numpy as np

A = np.matrix([
    [-1, 92.21, 75.84],
    [0.03, -7, -4.9],
    [0.03, -97.663, -92.752]
])

b = np.matrix([
    [0],
    [3],
    [-97]
])

In [4]:
A**-1*b

matrix([[31.57683805],
        [-3.92661553],
        [ 5.19053344]])

\begin{eqnarray}
  \begin{cases}
    P_0 = 31.58 & \\
    \alpha\ = -3.927 & \\
    \beta\  = 5.191
  \end{cases}
\end{eqnarray}